src/main/java/com/mongodb/hibernate/jdbc/UnsupportedOperationSQLException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import java.io.Serial;
+import java.sql.SQLException;
+
+/**
+ * The JDBC counterpart of {@link UnsupportedOperationException}.
+ *
+ * <p>Used only in {@link com.mongodb.hibernate.jdbc} package when implementing various JDBC API methods throwing
+ * {@link SQLException} invariably,
+ *
+ * <p>Hibernate expects the checked exception contract to go about business logic, so a {@link RuntimeException} like
+ * {@link UnsupportedOperationException} will break the contract and lead to unexpected behaviour.
+ *
+ * @see UnsupportedOperationException
+ */
+final class UnsupportedOperationSQLException extends SQLException {
```

Comment:
This exception class was not intended to be exposed to our user. It is internal class so not a part of the API. I need to dig more into Hibernate internal code details to know whether it will be exposed.

Point 3 is a valid point highly recommendable if we intend to expose this class to public. It is never used in Hibernate ORM internally.

Reply:
but I think you are right. This class should be exposed to the public. Need to rework on this. Your points are all valid.

Reply:
Found we'd better reuse existing `SQLFeatureNotSupportedException` JDBC exception class without reinventing wheels. Updated.

Reply:
I am glad you accepted `SQLFeatureNotSupportedException ` 👍 , as was suggested in https://github.com/mongodb/mongo-hibernate/pull/14#discussion_r1847301263.